### DISCUSSION TITLE
[brew doctor] Warning: Skipping **FORMULA** most recent version **VERSION** not installed

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#196] [Everyday usage] [answered]
brew doctor: Updating? FORMULA requires more _friendly tips_.
- update
- uninstall

Exec:
```sh
$ brew doctor
```
Output(_Ex:_ Warning dav1d: last version 0.7.1, current 0.7.0):
```sh
Warning: Skipping dav1d: most recent version 0.7.1 not installed
Warning: .....
```

Exec(_Ex_): 
```sh
$ brew doctor --ignore-upgrade
```
Output:
```sh
Note: ignore upgrade message. 
Warning: .....
```

>These are actually things that the user should be warned about. brew cleanup is failing to clean up folders because you haven't installed the latest version of the software. brew upgrade will install this latest version for you and then brew cleanup should be silent. @SMillerDev

**Comment by SMillerDev** [comment, marked as answer]
I'm not sure what the question is

**Reply by ghost** [reply]
thank's for reply @SMillerDev

Warning dav1d: Last Version 0.7.1, Current version 0.7.0.
replaced
Warning: dav1d: Most recent version 0.7.1 not installed.

Is it easier to understand.

see
https://github.com/amberframework/amber/issues/711#issuecomment-375888322

**Reply by SMillerDev** [reply]
I still don't understand.

Please fill out the following information, your current descriptions are way too brief to understand.
- What are you trying to do? (include example commands)
- Why are you trying to do this?
- What are you expecting to happen?
- What happened instead? (include logs of the problem)

**Reply by ghost** [reply]
thank's for reply @SMillerDev.

- What are you trying to do? (include example commands)
  + I have obsessive-compulsive disorder, I can ignore it or more messages when I don’t need it
    * `--ignore-warning-upgrade`
- Why are you trying to do this?
  + Seeing it working properly (ignoring unnecessary messages) can make me feel more secure
- What are you expecting to happen?
  + Ignore unnecessary messages
    * Use switcher (`--ignore-warning-upgrade`)
- What happened instead? (include logs of the problem)
  + Instead of such a message
    * `Warning: Skipping dav1d: most recent version 0.7.1 not installed`

**Reply by SMillerDev** [reply]
> I have obsessive-compulsive disorder, I can ignore it or more messages when I don’t need it 

Please give example brew commands that you ran. Currently all I know is why you think the warnings are a problem and what you want to pass to the command to fix them. I still don't know the following:


- What are you trying to do? (include example commands)
- Why are you trying to do this?
- What are you expecting to happen?
- What happened instead? (include logs of the problem)

**Reply by ghost** [reply]
~ % brew cleanup 
Warning: Skipping dav1d: most recent version 0.7.1 not installed
Warning: Skipping erlang: most recent version 23.1.4 not installed
Warning: Skipping ffmpeg: most recent version 4.3.1_4 not installed
Warning: Skipping freetype: most recent version 2.10.4 not installed
Warning: Skipping fribidi: most recent version 1.0.10 not installed
Warning: Skipping gcc: most recent version 10.2.0 not installed
Warning: Skipping gdk-pixbuf: most recent version 2.42.0 not installed
Warning: Skipping gettext: most recent version 0.21 not installed
Warning: Skipping glib: most recent version 2.66.2_1 not installed
Warning: Skipping gmp: most recent version 6.2.1 not installed
Warning: Skipping gnutls: most recent version 3.6.15 not installed
Warning: Skipping gsettings-desktop-schemas: most recent version 3.38.0 not installed
Warning: Skipping gtk+3: most recent version 3.24.23 not installed
Warning: Skipping harfbuzz: most recent version 2.7.2 not installed
Warning: Skipping leptonica: most recent version 1.80.0 not installed
Warning: Skipping libass: most recent version 0.15.0 not installed
Warning: Skipping libbluray: most recent version 1.2.1 not installed
Warning: Skipping libevent: most recent version 2.1.12 not installed
Warning: Skipping libmpc: most recent version 1.2.1 not installed
Warning: Skipping libsndfile: most recent version 1.0.30 not installed
Warning: Skipping libtiff: most recent version 4.1.0_1 not installed
Warning: Skipping libvorbis: most recent version 1.3.7 not installed
Warning: Skipping libvpx: most recent version 1.9.0 not installed
Warning: Skipping little-cms2: most recent version 2.11 not installed
Warning: Skipping mpfr: most recent version 4.1.0 not installed
Warning: Skipping mysql@5.7: most recent version 5.7.32 not installed
Warning: Skipping node@14: most recent version 14.15.1 not installed
Warning: Skipping p11-kit: most recent version 0.23.21 not installed
Warning: Skipping pango: most recent version 1.48.0 not installed
Warning: Skipping python@3.8: most recent version 3.8.6_2 not installed
Warning: Skipping rabbitmq: most recent version 3.8.9_1 not installed
Warning: Skipping rubberband: most recent version 1.9.0 not installed
Warning: Skipping srt: most recent version 1.4.2 not installed
Warning: Skipping unbound: most recent version 1.12.0_1 not installed
Warning: Skipping unrar: most recent version 6.0.2 not installed
Warning: Skipping wxmac: most recent version 3.0.5.1_1 not installed
Warning: Skipping x264: most recent version r3027_1 not installed
Warning: Skipping x265: most recent version 3.4_1 not installed

**Reply by SMillerDev** [reply]
These are actually things that the user should be warned about. `brew cleanup` is failing to clean up folders because you haven't installed the latest version of the software. `brew upgrade` will install this latest version for you and then `brew cleanup` should be silent.

**Comment by igroti** [comment]
I'm on M1 Mac.
I ran 
```
brew cleanup
brew update
brew cleanup
arch -x86_64 brew update
brew cleanup
arch -x86_64 brew upgrade
```
Multiple times, and keep getting the warning. I don't understand what I am expected to do.

Thanks.

**Reply by gromgit** [reply]
> I'm on M1 Mac.

So why are you trying to force Homebrew to run in "Intel mode" with
>```
>arch -x86_64 brew update
>arch -x86_64 brew upgrade
>```
?

**Reply by igroti** [reply]
I thought I was doing just the opposite with that, as I kept getting errors until I found that addition online.
What should I be running instead?

Ok, I think I have it working now (moved brew from usr/bin to opt/homebrew).

Would be nice if I could run something like "is brew setup correctly?". `brew doctor` does seem ok but I don't trust doctors. 😛
